cloud/amazon/s3_bucket.py

```diff
@@ -376,12 +437,13 @@ def main():
     state = module.params.get("state")
 
     if state == 'present':
-        create_bucket(connection, module, location)
+        create_bucket(connection, module, location, flavour=flavour)
     elif state == 'absent':
-        destroy_bucket(connection, module)
+        destroy_bucket(connection, module, flavour=flavour)
 
 from ansible.module_utils.basic import *
 from ansible.module_utils.ec2 import *
+import urlparse
```

Comment:
imports should be near the top of the file (For this one, it looks like urlparse is already imported: https://github.com/ansible/ansible-modules-extras/pull/1599/files#diff-d1353778df71e40405b333a88cdb9925L100 ).  from ansible.module_utils* is at the bottom because prior to 2.1, imports of module_utils was actually a preprocessor-like substitution.  So importing at the bottom helped preserve line numbers when debugging.  We'll be moving these to the top of files (and moving away from wildcard imports) as time goes on.